File: AudiobookManager/AudiobookManager.Scraping/Models/MetadataSearchServiceInfo.cs
namespace AudiobookManager.Scraping.Models;

public class MetadataSearchServiceInfo
{
    public string Name { get; set; }
    public bool Enabled { get; set; }
    public string? DisabledReason { get; set; }

    public MetadataSearchServiceInfo(string name, bool enabled, string? disabledReason = null)
    {
        Name = name;
        Enabled = enabled;
        DisabledReason = disabledReason;
    }
}

File: client/src/types/MetadataSearchServiceInfo.ts
import type { components } from "@/lib/api-types";
import type { Require } from "@/lib/dto";

// AudiobookManager.Scraping/Models/MetadataSearchServiceInfo.cs: name and enabled are
// non-nullable; disabledReason is genuinely optional.
export type MetadataSearchServiceInfo = Require<
  components["schemas"]["MetadataSearchServiceInfo"],
  "name" | "enabled"
>;
